### PR TITLE
Documentation for container-names annotation

### PR DESCRIPTION
**Pull request by fscellos**
@pavolloffay : missing documentation for multiple injection based on "instrumentation.opentelemetry.io/container-names"  #689

### CODE REVIEW

**Comment by fscellos** [comment]
e2e fail because correction of these tests is host by #875

**Review by pavolloffay — APPROVED** [review]

**Comment by pavolloffay** [comment]
Please rebase the PR and the CI should pass

File: README.md [hunk 1]
```diff
@@ -215,6 +215,36 @@ The possible values for the annotation can be
...
+
+If nothing else is specified, instrumentation is performed on the first container available in the pod spec.
+In some cases (for example in the case of the injection of an Istio sidecar) it becomes necessary to specify on which container(s) this injection must be performed.
+
+For this, it is possible to fine-tune the pod(s) on which the injection will be carried out.
+
+For this, we will use the "instrumentation.opentelemetry.io/container-names" annotation for which we will indicate one or more pod names (.spec.containers.name) on which the injection must be made:
+
+```yaml
+  template:
```
Comment (pavolloffay) [inline README.md:228]: Please provide the full yaml with kind and API version

File: README.md [hunk 2]
```diff
@@ -215,6 +215,36 @@ The possible values for the annotation can be
...
+    metadata:
+      labels:
+        app: my-pod-with-sidecar
+      annotations:
+        sidecar.opentelemetry.io/inject: "true"
```
Comment (pavolloffay) [inline README.md:233]: remove the sidecar annotaiton - keep this example focused only on a single feature

File: README.md [hunk 3]
```diff
@@ -215,6 +215,36 @@ The possible values for the annotation can be
...
+      containers:
+      - name: myapp
+        image: myImage1
+      - name: myapp2
+        image: myImage2
+      - name: myapp3
+        image: myImage3
+```
+
+In the above case, myapp and myapp2 containers will be instrumented, myapp3 will not.
```
Comment (pavolloffay) [inline README.md:246]: Use code formatting for `myapp`, `myapp2` and `myapp3`

File: README.md [hunk 4]
```diff
@@ -215,6 +215,36 @@ The possible values for the annotation can be
...
 * `"false"` - do not inject
 
+#### Multi-container pods
+
+If nothing else is specified, instrumentation is performed on the first container available in the pod spec.
+In some cases (for example in the case of the injection of an Istio sidecar) it becomes necessary to specify on which container(s) this injection must be performed.
+
+For this, it is possible to fine-tune the pod(s) on which the injection will be carried out.
+
+For this, we will use the "instrumentation.opentelemetry.io/container-names" annotation for which we will indicate one or more pod names (.spec.containers.name) on which the injection must be made:
```
Comment (pavolloffay) [inline README.md:225]: Use code formatting for the annotation and `.spec.containers.name`

File: README.md [hunk 5]
```diff
@@ -215,6 +215,44 @@ The possible values for the annotation can be
...
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment-with-sidecar
+spec:
+  selector:
+    matchLabels:
+      app: my-pod-with-sidecar
```
Comment (pavolloffay) [inline README.md:235]: there is no sidecar :)

Comment (fscellos) [reply README.md:235]: Not really :-).. I correct it ;-)

Comment (fscellos) [reply README.md]: Done.

**Pull request merged**